pkg/runtime/reconciler.go

```diff
@@ -507,12 +507,17 @@ func (r *resourceReconciler) patchResourceStatus(
 		latest.RuntimeObject(),
 		client.MergeFrom(desired.DeepCopy().RuntimeObject()),
 	)
-	rlog.Exit("kc.Patch (status)", err)
-	if err != nil {
-		return err
+	if err == nil {
+		rlog.Debug("patched resource status")
+	} else if apierrors.IsNotFound(err) {
+		rlog.Debug("skipping resource status patching because " +
+			"the resource no longer exists")
```

Comment:
I would just remove the above debug log message entirely. :) It's not particularly useful and kind of reminds me of the error message that was originally problematic